salt/transport/zeromq.py

```diff
@@ -754,14 +754,18 @@ def handle_message(self, stream, payload):
             stream.send(self.serial.dumps(self._auth(payload['load'])))
             raise tornado.gen.Return()
 
-        # TODO: test
+        def _return_error(msg, enc=payload.get('enc', 'clear')):
+            if enc != 'clear':
+                msg = self.crypticle.dumps(msg)
+            stream.send(self.serial.dumps(msg))
```

Comment:
Please suggest a better place to put this helper function.

Reply:
Probably the helper function should also be used in other cases: eg line 744 and 748

Reply:
Since such double-dumps is a very specific case used only here I'd suggest to leave it here.